components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/AppRoleMappingConfig.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.common.model;
+
+import org.apache.axiom.om.OMElement;
+
+import java.io.Serializable;
+import java.util.Iterator;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+
+/**
+ * Application role mapping type of application for an IdP.
+ */
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlRootElement(name = "ApplicationRoleMappingConfig")
+public class AppRoleMappingConfig implements Serializable {
+
+    private static final long serialVersionUID = -1870576785207238676L;
+    private static final String IDP_NAME = "IdPName";
+    private static final String USE_APP_ROLE_MAPPINGS = "UseAppRoleMappings";
+
+    @XmlElement(name = IDP_NAME)
+    private String idPName;
+
+    @XmlElement(name = USE_APP_ROLE_MAPPINGS)
+    private boolean useAppRoleMappings;
+
+    /**
+     * Build ApplicationRoleMappingConfig from OMElement.
+     *
+     * @param applicationRoleMappingConfigOM OMElement
+     * @return ApplicationRoleMappingConfig
+     */
```

Comment:
Shall we add some more description on this?

Reply:
Addressed in eb2c437f152c27dff29633549de3b77457525dd4